src/integrations/ga/SplitToGa.js

```diff
@@ -1,10 +1,108 @@
+    switch (type) {
+      case SPLIT_IMPRESSION:
+        return {
+          hitType: 'event',
+          eventCategory: 'split-impression',
+          eventAction: payload.impression.feature,
+          eventLabel: payload.impression.treatment,
+          nonInteraction: true,
+        };
+      case SPLIT_EVENT:
+        return {
+          hitType: 'event',
+          eventCategory: 'split-event',
+          eventAction: payload.eventTypeId,
+          eventValue: payload.value,
+          nonInteraction: true,
+        };
+    }
+    return null;
+  }
+
+  static getGa() {
+    return typeof window !== 'undefined' ? window[window['GoogleAnalyticsObject'] || 'ga'] : undefined;
+  }
+
+  /**
+   * Validates if a given object is a UniversalAnalytics.FieldsObject instance, and logs a warning if not.
+   * It checks that the object contains a `hitType`, since it is the minimal field required to send the hit
+   * and avoid the GA error `No hit type specified. Aborting hit.`.
+   * Other validations (e.g., an `event` hitType must have a `eventCategory` and `eventAction`) are handled
+   * and logged (as warnings or errors depending the case) by GA debugger, but the hit is sent anyway.
+   *
+   * @param {UniversalAnalytics.FieldsObject} fieldsObject object to validate.
+   * @returns {boolean} Whether the data instance is a valid FieldsObject or not.
+   */
+  static validateFieldsObject(fieldsObject) {
+    if (fieldsObject && fieldsObject.hitType)
+      return true;
+
+    log.warn('your custom mapper returned an invalid FieldsObject instance. It must be an object with at least a `hitType` field.');
+    return false;
+  }
+
+  constructor(options) {
+
+    // Check if `ga` object is available
+    if (typeof SplitToGa.getGa() !== 'function') {
+      // @TODO review the following warning message
+      log.warn('`ga` command queue not found. No hits will be sent.');
+      // Return an empty object to avoid creating a SplitToGa instance
+      return {};
+    }
+
+    this.filter = options && typeof (options.filter) === 'function' ?
+      options.filter :
+      SplitToGa.defaultFilter;
+
+    // @TODO Should we check something else about `configObject.impressionMapper`?
+    // It doesn't matter, because if the returned object is not a GA fieldsObject or string, ga send command will do nothing.
+    this.mapper = options && typeof (options.mapper) === 'function' ?
+      options.mapper :
+      SplitToGa.defaultMapper;
+
+    this.trackerNames = options && Array.isArray(options.trackerNames) ?
+      // We strip off duplicated values if we received a `trackerNames` param.
+      // We don't warn if a tracker does not exist, since the user might create it after the SDK is initialized.
+      // Note: GA allows to create and get trackers using a string or number as tracker name, and does nothing if other types are used.
+      uniq(options.trackerNames) :
+      SplitToGa.defaultTrackerNames;
+  }
+
+  queue(data) {
+    try {
```

Comment:
Don't just wrap everything in this but only what's necessary. Unless this changed (it's been a couple of years since I looked into it) there are less optimizations possible on the machine code generated inside a try..catch block